src/app/api/workflow-states/workflowStates.service.ts

```diff
@@ -11,14 +11,33 @@ class WorkflowStatesService extends BaseService {
     const policyGate = new PoliciesService(this.user)
     policyGate.authorize(UserAction.Read, Resource.Tasks)
 
-    const filters = { where: { workspaceId: this.user.workspaceId } }
-    let workflowStates = await this.db.workflowState.findMany(filters)
+    let workflowStates: WorkflowStateResponse[] = await this.db.$queryRaw`
+    SELECT * FROM "WorkflowStates"
+    WHERE "workspaceId" = ${this.user.workspaceId}
+    ORDER BY 
+      CASE 
+        WHEN "key" = 'todo' THEN 1
+        WHEN "key" = 'inProgress' THEN 2
+        WHEN "key" = 'completed' THEN 3
+        ELSE 4
+      END;
+  `
```

Comment:
Key can be anything based on the name of the workflow state.. If i named 'todo' something like "Pending", this logic would all fall apart.

Please order using the Workflowstate type instead 
![image](https://github.com/copilot-platforms/tasks-app/assets/59971845/9bd5ba33-a974-4918-942e-fc16f408d400)